src/overlays/actors/ovl_En_Pamera/z_en_pamera.c

```diff
@@ -22,135 +104,791 @@ const ActorInit En_Pamera_InitVars = {
+    if (pathID >= 0) {
+        this->pathPoints = Lib_SegmentedToVirtual(path->points);
+        this->pathIndex = 0;
+        this->pathPointsCount = path->count;
+        this->pathID = path->unk1;
+    }
+}
+
+void func_80BD9338(EnPamera* this, GlobalContext* globalCtx) {
+    Actor* actor;
+
+    func_80BD84F0(this, globalCtx);
+    actor = this->actor.child;
+    if ((actor != NULL) && (actor->id == ACTOR_EN_DOOR)) {
+        ((EnDoor*)actor)->unk1A7 = -0x32;
+    }
+}
+
+void func_80BD9384(EnPamera* this, GlobalContext* globalCtx) {
+    if (this->unk_316 < 0x28) {
+        this->unk_316++;
+    } else {
+        this->unk_312 += 1;
+        if (this->unk_312 >= 3) {
+            this->unk_312 = 0;
+            this->unk_316 = 0;
+        }
+    }
+}
+
+void func_80BD93CC(EnPamera* this, s16 arg1, s16 arg2) {
+    this->unk_310 = arg1;
+    this->unk_314 = arg2;
+}
+
+void func_80BD93F4(EnPamera* this, GlobalContext* globalCtx) {
+    if ((this->actionFunc == func_80BD8B70) || (this->actionFunc == func_80BD8DB0) ||
+        (this->actionFunc == func_80BD8964) || (this->actionFunc == func_80BD8A7C)) {
+        if (this->skelAnime.animCurrentSeg == &D_06008AE0) {
+            if (func_801378B8(&this->skelAnime, 9.0f) || func_801378B8(&this->skelAnime, 18.0f)) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_PAMERA_WALK);
+            }
+        } else if ((this->skelAnime.animCurrentSeg == &D_06008E38) &&
+                   (func_801378B8(&this->skelAnime, 2.0f) || func_801378B8(&this->skelAnime, 6.0f))) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_PAMERA_WALK);
+        }
+    }
+}
+
+void func_80BD94E0(EnPamera* this, GlobalContext* globalCtx) {
+    if ((this->actionFunc != func_80BD8B70) && (this->actionFunc != func_80BD8964) &&
+        (this->actionFunc != func_80BD909C) && (this->actionFunc != func_80BD8D1C) &&
+        ((this->actionFunc != func_80BD8DB0) || (this->actor.speedXZ == 3.0f))) {
+        func_800E9250(globalCtx, &this->actor, &this->limb15Rot, &this->limb8Rot, this->actor.focus.pos);
+    } else {
+        func_800E8F08(&this->limb15Rot, &this->limb8Rot);
+    }
+}
+
+void EnPamera_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnPamera* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    func_80BD90AC(this, globalCtx);
+    func_80BD9384(this, globalCtx);
+    func_80BD94E0(this, globalCtx);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    func_80BD93F4(this, globalCtx);
+}
+
+s32 EnPamera_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                              Actor* thisx) {
+    EnPamera* this = THIS;
+
+    if (limbIndex == 9) {
+        rot->x += this->limb15Rot.y;
+        rot->z += this->limb15Rot.x;
+    }
```

Comment:
```suggestion
    if (limbIndex == PAMERA_LIMB_HAIR) {
        rot->x += this->limb15Rot.y;
        rot->z += this->limb15Rot.x;
    }
```
Is there any particular reason why it's `limb15Rot` and not `limb9Rot`?

Reply:
Probably because I was looking at an actor that looked similar at the time :sweat_smile: